backend/src/db/repositories/UserRepository.ts

```diff
@@ -25,6 +26,53 @@ export default class UserRepository {
         return await this.db.fetchOne<UserRaw>('select * from users where username=:username', {username: username});
     }
 
+    // TODO: need to decide if we want to encrypt email addresses
+    async getUserByEmail(email: string): Promise<UserRaw | undefined> {
+        return await this.db.fetchOne<UserRaw>('select * from users where email=:email', {email});
+    }
+
+    async generateAndSavePasswordResetForUser(userId: number): Promise<string | undefined> {
+        const code = crypto.createHash('md5').update(Math.random().toString() + userId).digest('hex');
```

Comment:
```suggestion
        const code = crypto.createHash('sha256').update(Math.random().toString() + userId).digest('hex');
```

Индустрия давно отказалась от md5 и гонит поганой метлой sha1, так что, во избежание будущих сломов предлагаю золотой стандарт текущего дня.